plugin/mmdb/setup.go

```diff
@@ -0,0 +1,94 @@
+package mmdb
+
+import (
+	"net"
+	"path/filepath"
+
+	"github.com/coredns/caddy"
+	"github.com/coredns/coredns/core/dnsserver"
+	"github.com/coredns/coredns/plugin"
+	"github.com/oschwald/geoip2-golang"
+)
+
+const pluginName = "mmdb"
+
+func init() { plugin.Register(pluginName, setup) }
+
+func setup(c *caddy.Controller) error {
+	mmdb, err := mmdbParse(c)
+	if err != nil {
+		return plugin.Error(pluginName, err)
+	}
+
+	dnsserver.GetConfig(c).AddPlugin(func(next plugin.Handler) plugin.Handler {
+		mmdb.Next = next
+		return mmdb
+	})
+
+	return nil
+}
+
+// probingIP is an arbitrary internet IP used to discover database types through database lookups.
+var probingIP = net.ParseIP("81.2.69.142")
+
+func mmdbParse(c *caddy.Controller) (*Mmdb, error) {
+	dbPaths := make(map[string]bool)
+	for c.Next() {
+		if !c.NextArg() {
+			return nil, c.ArgErr()
+		}
+		path := c.Val()
+		if c.NextArg() { // There shouldn't be any more arguments.
+			return nil, c.ArgErr()
+		}
+		if dbPaths[path] {
+			return nil, c.Errf("repeating mmdb path %q is not allowed", path)
+		}
+		dbPaths[path] = true
+		c.NextBlock()
+	}
+
+	// TODO(snebel29): Warn or Fail when configuring overlapping db/fields?
+	// TODO(snebel29): Fail if we open more than once the same dbType?
+	dbs := make([]mmdb, 0, len(dbPaths))
+
+	for path := range dbPaths {
+		reader, err := geoip2.Open(path)
+		if err != nil {
+			return nil, plugin.Error(pluginName, c.Errf("failed to open database file: %v", err))
```

Comment:
plugin.Error is already called when this function returns

Reply:
ok, will double check and remove redundant error.